doc/modules/preprocessing.rst

```diff
@@ -445,28 +445,97 @@ values, either using the mean, the median or the most frequent value of
+indicator matrix is shown, then shape of the transformed input is
+(n_samples, n_features_new + len(imputed_features_) ::
 
     >>> import numpy as np
     >>> from sklearn.preprocessing import Imputer
     >>> imp = Imputer(missing_values='NaN', strategy='mean', axis=0)
-    >>> imp.fit([[1, 2], [np.nan, 3], [7, 6]])
-    Imputer(axis=0, copy=True, missing_values='NaN', strategy='mean', verbose=0)
+    >>> imp.fit([[1, 2], [np.nan, 3], [7, 6]])  # doctest: +NORMALIZE_WHITESPACE
+    Imputer(add_missing_indicator=False, axis=0, copy=True, missing_values='NaN',
+        strategy='mean', verbose=0)
     >>> X = [[np.nan, 2], [6, np.nan], [7, 6]]
     >>> print(imp.transform(X))                           # doctest: +ELLIPSIS
     [[ 4.          2.        ]
      [ 6.          3.666...]
      [ 7.          6.        ]]
+    >>> imp_with_in = Imputer(missing_values='NaN', strategy='mean', axis=0,add_missing_indicator=True)
+    >>> imp_with_in.fit([[1, 2], [np.nan, 3], [7, 6]])
+    Imputer(add_missing_indicator=True, axis=0, copy=True, missing_values='NaN',
+        strategy='mean', verbose=0)
+    >>> print(imp_with_in.transform(X))                           # doctest: +ELLIPSIS
+    [[ 4.          2.          1.          0.        ]
+     [ 6.          3.66666667  0.          1.        ]
+     [ 7.          6.          0.          0.        ]]
+    >>> print(imp_with_in.imputed_features_)
+    [0 1]
+
+The following case is when a feature has all missing values and the feature gets
+discarded upon transform and won't be shown in ``imputed_features_``::
+
+    >>> import numpy as np
+    >>> from sklearn.preprocessing import Imputer
+    >>> imp = Imputer(missing_values='NaN', strategy='mean', axis=0)
+    >>> imp.fit([[np.nan, 2], [np.nan, 3], [np.nan, 6]]) # doctest: +NORMALIZE_WHITESPACE
+    Imputer(add_missing_indicator=False, axis=0, copy=True, missing_values='NaN',
+        strategy='mean', verbose=0)
+    >>> X = [[np.nan, 2], [6, np.nan], [7, 6]]
+    >>> print(imp.transform(X))                           # doctest: +ELLIPSIS
+    [[ 2.        ]
+     [ 3.66666667]
+     [ 6.        ]]
+    >>> imp_with_in = Imputer(missing_values='NaN', strategy='mean', axis=0, add_missing_indicator=True)
+    >>> imp_with_in.fit([[np.nan, 2], [np.nan, 3], [np.nan, 6]])
+    Imputer(add_missing_indicator=True, axis=0, copy=True, missing_values='NaN',
+        strategy='mean', verbose=0)
+    >>> print(imp_with_in.transform(X))                           # doctest: +ELLIPSIS
+    [[ 2.          0.        ]
+     [ 3.66666667  1.        ]
+     [ 6.          0.        ]]
+    >>> print(imp_with_in.imputed_features_)
+    [1]
+
+In case the imputed value is to be calculated using the rows (axis 1) then
+transform can be directly called since the value is calculated across sample.
+Also if the indicator matrix is shown, then shape of the transformed input is
+(n_samples, n_features + len(imputed_features_) ::
```

Comment:
Again, I consider this extraneous. The narrative doc isn't there for testing, it's there for outlining motivation and usage in a didactic manner.